apps/keyboard/js/settings/user_dictionary_list_panel.js

```diff
@@ -106,16 +121,32 @@ UserDictionaryListPanel.prototype.handleEvent = function(evt) {
   }
 };
 
-UserDictionaryListPanel.prototype._appendList = function(word) {
-  var wordElem = document.createElement('a');
-  wordElem.href = '#' + word;
-  wordElem.textContent = word;
+// Re-arrange the DOM elements for the list of words. Create related dom element
+// if needed.
+UserDictionaryListPanel.prototype._rearrangeList = function(wordList) {
+  wordList.forEach(word => {
+    word = word.trim();
+
+    var wordLIElem;
 
-  var li = document.createElement('li');
-  li.appendChild(wordElem);
-  this._listContainer.appendChild(li);
+    if (!(word in this._wordDomMap)) {
+      var wordElem = document.createElement('a');
+      wordElem.href = '#' + word;
+      wordElem.textContent = word;
 
-  this._domWordMap.set(wordElem, word);
+      wordLIElem = document.createElement('li');
+      wordLIElem.appendChild(wordElem);
+
+      this._domWordMap.set(wordElem, word);
+      this._wordDomMap[word] = wordElem;
+    } else {
+      wordLIElem = this._wordDomMap[word].parentNode;
```

Comment:
This `parentNode` look strange to me. After some digging, I realized you are save the reference to `<a>` and we need to append the parent node (`<li>`) here, and the reason you save `<a>` is because you are using that reference in `handleEvent`.

Let me proposed an alternative here:
1. If we could get rid of `<a>` without mess up with the style, please do.
2. If we couldn't, how about use `pointer-event: none;` to make sure we are always getting `evt.target` points to `<li>`? That way, we could limit the `<a>` as a detail to be disregarded in JS.

If either thing is too much to do in this patch, for get about it.